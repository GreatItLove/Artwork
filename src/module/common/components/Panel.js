import React from 'react';
import classNames from 'classnames';

const Panel = props => {
  const panelClass = classNames({
    'readmin-panel': true,
    'body-text-center': props.center
  });

  const panelHeadingClassName = classNames({
    'panel-heading': true,
    [props.panelHeadingClassName]: props.panelHeadingClassName
  });

  return (
    <div className={panelClass} style={props.style}>
      <div className={panelHeadingClassName} style={props.headingStyle}>
        <h5 style={props.titleStyle}>
          {props.title}
          {props.refresh}
        </h5>
        {props.rightElement}
      </div>
      <div className="panel-body">{props.children}</div>
      <div className="pb-2">{props.actions}</div>
    </div>
  );
};

export default Panel;
