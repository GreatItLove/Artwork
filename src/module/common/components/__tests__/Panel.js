import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Panel from '../Panel';

const RightElement = () => <div>Right Element</div>;
describe('Panel', () => {
  it('Snapshot of Panel component', () => {
    const wrapper = shallow(<Panel />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Snapshot of Panle with right element', () => {
    const wrapper = shallow(<Panel rightElement={<RightElement />} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
