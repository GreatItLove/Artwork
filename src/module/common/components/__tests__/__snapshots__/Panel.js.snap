// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Snapshot of Panel component 1`] = `
<div
  className="readmin-panel"
>
  <div
    className="panel-heading"
  >
    <h5 />
  </div>
  <div
    className="panel-body"
  />
</div>
`;

exports[`Panel Snapshot of Panle with right element 1`] = `
<div
  className="readmin-panel"
>
  <div
    className="panel-heading"
  >
    <h5 />
    <RightElement />
  </div>
  <div
    className="panel-body"
  />
</div>
`;
